src/main/java/org/broadinstitute/hellbender/utils/variant/writers/GVCFWriter.java

```diff
@@ -59,46 +37,12 @@
      * @param defaultPloidy    the assumed ploidy for input variant context without one.
      */
     public GVCFWriter(final VariantContextWriter underlyingWriter, final List<Integer> gqPartitions, final int defaultPloidy) {
-        this.underlyingWriter = Utils.nonNull(underlyingWriter);
-        this.gqPartitions = parsePartitions(gqPartitions);
-        this.defaultPloidy = defaultPloidy;
+        this(underlyingWriter, new GVCFBlockCombiner(gqPartitions, defaultPloidy));
     }
 
-    /**
-     * Create {@link HomRefBlock}s which will collectively accept variants of any genotype quality
-     *
-     * Each individual block covers a band of genotype qualities with the splits between bands occurring at values in {@code gqPartitions}.
-     * There will be {@code gqPartitions.size() +1} bands produced covering the entire possible range of genotype qualities from 0 to {@link VCFConstants#MAX_GENOTYPE_QUAL}.
-     *
-     * @param gqPartitions proposed GQ partitions
-     * @return a list of HomRefBlocks accepting bands of genotypes qualities split at the points specified in gqPartitions
-     */
-    @VisibleForTesting
-    static RangeMap<Integer,Range<Integer>> parsePartitions(final List<Integer> gqPartitions) {
-        Utils.nonEmpty(gqPartitions);
-        Utils.containsNoNull(gqPartitions, "The list of GQ partitions contains a null integer");
-        final RangeMap<Integer, Range<Integer>> result = TreeRangeMap.create();
-        int lastThreshold = 0;
-        for (final Integer value : gqPartitions) {
-            if (value < 0) {
-                throw new IllegalArgumentException("The list of GQ partitions contains a non-positive integer.");
-            } else if (value > MAX_GENOTYPE_QUAL + 1) {
-                throw new IllegalArgumentException(String.format("The value %d in the list of GQ partitions is greater than VCFConstants.MAX_GENOTYPE_QUAL + 1 = %d.", value, MAX_GENOTYPE_QUAL + 1));
-            } else if (value < lastThreshold) {
-                throw new IllegalArgumentException(String.format("The list of GQ partitions is out of order. Previous value is %d but the next is %d.", lastThreshold, value));
-            } else if (value == lastThreshold) {
-                throw new IllegalArgumentException(String.format("The value %d appears more than once in the list of GQ partitions.", value));
-            }
-
-            result.put(Range.closedOpen(lastThreshold, value), Range.closedOpen(lastThreshold, value));
-            lastThreshold = value;
-        }
-
-        if (lastThreshold <= MAX_GENOTYPE_QUAL) {
-            result.put(Range.closedOpen(lastThreshold, MAX_GENOTYPE_QUAL + 1), Range.closedOpen(lastThreshold,MAX_GENOTYPE_QUAL + 1));
-        }
-
-        return result;
+    public GVCFWriter(final VariantContextWriter underlyingWriter, GVCFBlockCombiner combiner) {
```

Comment:
This public constructor seems redundant. If there is an actual use for it, it should have javadoc describing that.

Reply:
Combined into a single constructor